streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java

```diff
@@ -79,25 +88,27 @@
 
     private static final String APPID = "restore-test";
 
-    @ClassRule
-    public static final EmbeddedKafkaCluster CLUSTER =
-            new EmbeddedKafkaCluster(NUM_BROKERS);
+    public final EmbeddedKafkaCluster cluster = new EmbeddedKafkaCluster(NUM_BROKERS);
```

Comment:
I made the cluster non-static because there was poor isolation between the tests and, for example, `setCommittedOffset` would often fail. This was just the "lazy" solution, we could also make sure to give the mall unique application ids

Reply:
I see. Maybe making it a Rule instead of a ClassRule is the solution you're looking for?

But yes, there's a middle ground, which is to use uniquely named resources for each test, for which we have `org.apache.kafka.streams.integration.utils.IntegrationTestUtils#safeUniqueTestName`.

Reply:
What's the advantage of a Rule over just...nothing?

Mostly I just didn't want to hard code a bunch of names in myself. But `safeUniqueTestName` sounds like what I need

Reply:
Nevermind, I take it the answer is "it calls `start` for you since the embedded cluster extends `ExternalResource` ?"

Reply:
I remember now, the main reason for the poor test isolation was different tests writing various amounts of data to the different topics at specific offsets. Honestly I'm not sure why this wasn't causing problems before 🤔 
Although...I notice that the first test writes 10000 keys to the input topic starting at offset 0, while the second test to write to the input topic does so starting at offset 10000. Clearly I should have looked at all the other tests in the class, counted them up, and then just started writing data at the highest offset